app/services/importers/doc/profile.rb

```diff
@@ -15,22 +16,17 @@ def perform
           bar.increment!
           data = line.unpack(field_pattern).map(&:squish)
 
-          save_profile(data)
+          @profiles << save_profile(data)
         end
+
+        ::Doc::Profile.upsert_all(@profiles, unique_by: :doc_number)
```

Comment:
this one is correct. If you notice in the other methods they use `find_or_initialize_by` inside the save_profile method